crates/swc_ecma_minifier/tests/terser/compress/issue_597/beautify_off_1/output.js

```diff
@@ -1,2 +1,2 @@
 var NaN1;
```

Comment:
It should be `NaN`

Reply:
It can't be `NaN`. In L2 there's reference to global `NaN`, the only thing can do is to rename this to avoid shadowing.

Reply:
```
❯ node
Welcome to Node.js v18.18.2.
Type ".help" for more information.
> var NaN;
undefined
> console.log(
...     null,
...     undefined,
...     Infinity,
...     NaN,
...     Infinity * undefined,
...     Infinity.toString(),
...     NaN.toString(),
...     (Infinity * undefined).toString()
... );
null undefined Infinity NaN NaN Infinity NaN NaN
undefined
```

Reply:
Shadowing can have different outcomes, such as taking effect, causing silent errors, or resulting in an error message. Due to the lack of information, we can only determine the result of shadowing at runtime. The only correct approach is to avoid renaming.

Reply:
Are you talking about the ESM mode?

Basically, the input is https://github.com/Austaras/swc/blob/16eee5ce3f12154ea8f5e3e62d6b25805a3504b6/crates/swc_ecma_minifier/tests/terser/compress/issue_597/beautify_off_1/input.js and when I tried executing code, the result was different from the result of executing the original input file

Reply:
Node repl is not esm, so this is not relate to this PR, which only handles output in esm.

Reply:
> Are you talking about the ESM mode?
> 

Yes.

Reply:
But in non-strict non-esm js, the end result should be
```js
var NaN1;
console.log(null, void 0, 1 / 0, NaN, NaN, (1 / 0).toString(), NaN.toString(), NaN.toString());
```
which is consistent between nodejs and browser.

Reply:
Oh, and terser would generate
```js
var NaN1;
console.log(null, void 0, 1 / 0, NaN1, NaN, (1 / 0).toString(), NaN1.toString(), NaN.toString());
```
when `module: false`